### ISSUE TITLE
Test suite

### ISSUE THREAD

**Issue by lfos** [poolpOrg/filter-senderscore#8]
Would you be interested in a simple test suite that runs `filter-senderscore` through a bunch of scenarios and validates the output? If so, any preferences when it comes to test frameworks? Would you prefer something that's TAP-compatible?

**Comment by poolpOrg** [comment]
yes, I'd be interested :-)

no preference for the framework as long as its not overkill, any idea in mind ?

**Comment by lfos** [comment]
There are many options. The "least overkill" option is to write own small framework using some small Shell scripts (and maybe Go or Python). I've used Sharness before which is TAP-compatible and very lightweight yet powerful -- but it's GPL-licensed. Then, there are other more popular frameworks like Google Test which are more powerful but less lightweight.

**Comment by lfos** [comment]
I implemented a minimalistic TAP-compatible POSIX shell solution with minimal overhead (~10 lines of general purpose code) and no external dependencies.

It requires a small change to `filter-senderscore` though. I have a patch that is based on the other PRs that are currently still open (#13 and #16). Will submit another PR once they're merged, and then implement some basic test cases.

**Comment by poolpOrg** [comment]
they are merged now, thanks.

curious about your shell script now :-)

**Comment by lfos** [comment]
Great! I submitted the preparatory patch that I announced before. It adds a way to make the output of the script deterministic which is a requirement for the test suite (unless we want to make the test suite overly complex). Once that is merged, I'd be happy to submit a PR for the test framework and the test cases!
